src/traits.rs

```diff
@@ -15,9 +18,39 @@ impl TextSized for &'_ str {
     }
 }
 
+impl<D> TextSized for &'_ D
+where
+    D: Deref<Target = str>,
+{
+    #[inline]
+    fn text_size(self) -> TextSize {
+        self.deref().text_size()
+    }
+}
+
 impl TextSized for char {
     #[inline]
     fn text_size(self) -> TextSize {
         TextSize(self.len_utf8() as u32)
     }
 }
+
+// assertion shape from static_assertions::assert_impl_all!
+const _: fn() = || {
+    use std::borrow::Cow;
+
+    fn assert_impl<T: TextSized>() {}
+
+    assert_impl::<&String>();
+    assert_impl::<&Cow<str>>();
+
+    struct StringLike {}
+    impl Deref for StringLike {
+        type Target = str;
+        fn deref(&self) -> &str {
+            unreachable!()
+        }
+    }
+
+    assert_impl::<&StringLike>();
+};
```

Comment:
I'd rather write a real test with local variables, as that's what we actually care about is whether `TextSize::of(&String::new())` works, and not about passing an arg into `T: TextSize`. And I'd do it as a test, to reduce the noise in the library. But no too strong opinion on both cases!

bors d+